Added some new UI features

danfinlay opened the pull request:
Me and @kumavis added some sweet new features:
- New visual keyboard shows what you're pressing, and what plays in a loop.
- You can now drag and drop sound links from the web onto those keys to assign those sounds to the keyboard.
- When changing layout, it is persisted in localStorage, and loaded back up on reload.

Commit:
Fixed some broken file links.

The index.js pointed at some files that were not in the samples folder.

I removed the dead links, moved the link hash into a json config file.

Commit:
Rudimentary keyboard styling

Commit:
Fixed styles for chrome

Commit:
add data-key to keyboard elements

danfinlay committed:
Merge pull request #1 from kumavis/keyboard

add data-key to keyboard elements

Commit:
highlight key

Commit:
keypress fixes

Commit:
clean

danfinlay committed:
Merge pull request #2 from kumavis/keyboard

Keyboard

Commit:
Fixed style for chrome again

Commit:
audio drop in works

Commit:
clean

Commit:
Factored out samples

Commit:
Moved configuration objects into configuration files.

Commit:
Cleaned up

Commit:
Fixed silly merge conflict

Commit:
Merged

Commit:
Added localStorage persisting of layout

Commit:
Restored data tags to html

Commit:
Fixed bug

Commit:
Fixed keyboard-key class for drop event

Commit:
Added initial baudio link receiving

Also includes compiling baudio files into functions.  Does not handle
playing of baudio audio, although now includes baudio module.

Commit:
baudio keybinding and playback

danfinlay committed:
Merge pull request #4 from kumavis/baudio

Baudio key binding

flyswatter commented:
Now supports dragging studio.substack links onto keys to turn functions into key-pressable percussion instruments.

This feature is currently interrupts/blocks wav files playing, and does not "restart" the sound on key press, these will be great things to enhance it with.

Commit:
Added transfer failure resilience

When initially requesting audio files, the the app will now retry each
file twice before throwing an error, and when this error occurs, I'm
throwing up an alert now, so the user knows why no instruments are
working.

Commit:
Added basic instructions

Commit:
Added simple readme

Commit:
Fixed unreliable buffer type error

Commit:
Fixed some key class names that prevented assigning some new sounds

Commit:
Added simple title to page

Commit:
Added page title
